UnityProject/Assets/Scripts/Managers/SubSceneManager/SubSceneManager.Client.cs

```diff
@@ -29,24 +35,91 @@ void MonitorServerSceneListOnClient()
 		}
 	}
 
-	IEnumerator LoadClientSubScene(SceneInfo sceneInfo)
+	IEnumerator LoadClientSubScene(SceneInfo sceneInfo, bool HandlSynchronising = true,
+		SubsceneLoadTimer SubsceneLoadTimer = null, bool OverrideclientIsLoadingSubscene = false)
 	{
 		if (sceneInfo.SceneType == SceneType.MainStation)
 		{
-			var clientLoadTimer = new SubsceneLoadTimer();
-			//calculate load time:
-			clientLoadTimer.MaxLoadTime = 10f;
-			clientLoadTimer.IncrementLoadBar($"Loading {sceneInfo.SceneName}");
-			yield return StartCoroutine(LoadSubScene(sceneInfo.SceneName, clientLoadTimer));
+			if (SubsceneLoadTimer == null)
+			{
+				SubsceneLoadTimer = new SubsceneLoadTimer();
+				//calculate load time:
+				SubsceneLoadTimer.MaxLoadTime = 10f;
+			}
+
+			SubsceneLoadTimer.IncrementLoadBar($"Loading {sceneInfo.SceneName}");
+			yield return StartCoroutine(LoadSubScene(sceneInfo.SceneName, SubsceneLoadTimer, HandlSynchronising));
 			MainStationLoaded = true;
-			yield return WaitFor.Seconds(0.1f);
-			UIManager.Display.preRoundWindow.CloseMapLoadingPanel();
+
 		}
 		else
 		{
-			yield return StartCoroutine(LoadSubScene(sceneInfo.SceneName));
+			if (SubsceneLoadTimer != null)
+			{
+				SubsceneLoadTimer.IncrementLoadBar($"Loading {sceneInfo.SceneName}");
+			}
+
+			yield return StartCoroutine(LoadSubScene(sceneInfo.SceneName, HandlSynchronising  :HandlSynchronising ));
+		}
+
+		if (OverrideclientIsLoadingSubscene == false)
```

Comment:
Why here and not just do it before the loop and after the loop? Why does this method need to know about this bool when it is only relevant for the code that uses this method

Reply:
because it's used for the sub scene manager, To prevent multiple scenes being loaded at once